### PR TITLE
Make managed project types explicit and distinct from manager names

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: analyzer/src/main/kotlin/PackageManager.kt [hunk 1]
```diff
@@ -54,12 +54,13 @@ typealias ManagedProjectFiles = Map<PackageManagerFactory, List<File>>
...
 /**
- * A class representing a package manager that handles software dependencies. The package manager is referred to by its
- * [managerName]. The analysis of any projects and their dependencies starts in the [analysisRoot] directory using the
+ * A class to represent a package manager of the given [managerName] that handles dependencies for the given
+ * [projectType]. The analysis of any projects and their dependencies starts in the [analysisRoot] directory using the
  * given general [analyzerConfig]. Per-repository configuration is passed in [repoConfig].
  */
 abstract class PackageManager(
     val managerName: String,
+    val projectType: String,
```
Comment (fviernau) [inline analyzer/src/main/kotlin/PackageManager.kt:63]: Is there any manager which may return multiple types? (Not sure about Pub, but maybe this just returns dependencies of different type?)

File: analyzer/src/main/kotlin/PackageManager.kt [hunk 2]
```diff
@@ -54,12 +54,13 @@ typealias ManagedProjectFiles = Map<PackageManagerFactory, List<File>>
 typealias ProjectResults = Map<File, List<ProjectAnalyzerResult>>
 
 /**
- * A class representing a package manager that handles software dependencies. The package manager is referred to by its
- * [managerName]. The analysis of any projects and their dependencies starts in the [analysisRoot] directory using the
+ * A class to represent a package manager of the given [managerName] that handles dependencies for the given
```
Comment (fviernau) [inline analyzer/src/main/kotlin/PackageManager.kt:57]: Q: For what purpose do we now use the manager name?

File: analyzer/src/main/kotlin/PackageManager.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt:63]: Not right now. I was thinking about this, too, but came to the conclusions that even for managers like Gradle or Bazel (which via plugins support multiple programming languages / ecosystems), you probably still would want to use "Gradle" or "Bazel" here, but different types for the IDs of packages.

File: analyzer/src/main/kotlin/PackageManager.kt (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt:57]: A: For referring to it from configuration, e.g. to enable / disable it from the CLI.

Comment (fviernau) [reply analyzer/src/main/kotlin/PackageManager.kt:57]: Thanks!

**Review by fviernau — APPROVED** [review]

**Pull request merged**